src/formatDate.js

```diff
@@ -9,6 +9,35 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const oldSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+
+  const spliteOldDate = date.split(oldSeparator);
+
+  const newDate = Array(3);
+
+  for (let i = 0; i < spliteOldDate.length; i++) {
+    const oldPart = fromFormat[i];
+
+    let index = toFormat.indexOf(oldPart);
```

Comment:
This line assumes that each part of `fromFormat` will exist in `toFormat`, which is a valid assumption based on the task description. However, the code does not handle the case where `oldPart` is not found in `toFormat`, as `indexOf` would return `-1`. This can potentially assign a date part to an incorrect index. To make the code more robust, consider checking if `index` is not `-1` before assigning to `newDate[index]`.